core/src/main/java/org/mobilitydata/gtfsvalidator/model/SampleNotice.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2021 MobilityData IO
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.mobilitydata.gtfsvalidator.model;
+
+import com.google.gson.annotations.Expose;
+import com.google.gson.internal.LinkedTreeMap;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import org.mobilitydata.gtfsvalidator.notice.SeverityLevel;
+
+/**
+ * Used to deserialize a validation report. It is used to store information about one type of notice
+ * encountered in a validation report: error code, severity level, the total number of notices
+ * related to the error code and a list of notice contexts (which provides additional information
+ * about each notice.
+ */
+public class SampleNotice {
```

Comment:
> Maybe `NoticeReport`? You'd then have a `ValidationReport` that contained multiple `NoticeReports`.

👍🏾 

> IMHO `SampleNotice` isn't very clear on it's purpose, and it's confusing that a single `SampleNotice` contains multiple `sampleNotices`.


Let's consider the following validation report: 
```json 
{
  "notices": [
    {
      "code": "duplicate_route_name",
      "severity": "WARNING",
      "totalNotices": 43,
      "sampleNotices": [
        {
          "duplicatedField": "route_short_name",
          "csvRowNumber": 35,
          "routeId": "23883_3"
        },
        {
          "duplicatedField": "route_short_name",
          "csvRowNumber": 68,
          "routeId": "22837_3"
        },
      ]
    },
    {
      "code": "invalid_url",
      "severity": "ERROR",
      "totalNotices": 3,
      "sampleNotices": [
        {
          "filename": "agency.txt",
          "csvRowNumber": 20,
          "fieldName": "agency_url",
          "fieldValue": "https://svendborg-havn.dk/færger/"
        },
        {
          "filename": "agency.txt",
          "csvRowNumber": 30,
          "fieldName": "agency_url",
          "fieldValue": "https://www.svendborg-havn.dk/færger-fra-svendborg"
        }
      ]
    },
    {
      "code": "leading_or_trailing_whitespaces",
      "severity": "ERROR",
      "totalNotices": 83,
      "sampleNotices": [
        {
          "filename": "stops.txt",
          "csvRowNumber": 782,
          "fieldName": "stop_name",
          "fieldValue": "Skørping Station "
        },
        {
          "filename": "stops.txt",
          "csvRowNumber": 783,
          "fieldName": "stop_name",
          "fieldValue": "Skørping Station "
        }
      ]
    },
    {
      "code": "unexpected_enum_value",
      "severity": "WARNING",
      "totalNotices": 91,
      "sampleNotices": [
        {
          "filename": "routes.txt",
          "csvRowNumber": 351,
          "fieldName": "route_type",
          "fieldValue": 109
        },
        {
          "filename": "routes.txt",
          "csvRowNumber": 367,
          "fieldName": "route_type",
          "fieldValue": 109
        }
      ]
    },
    {
      "code": "unknown_column",
      "severity": "INFO",
      "totalNotices": 4,
      "sampleNotices": [
        {
          "filename": "transfers.txt",
          "fieldName": "from_route_id",
          "index": 5
        },
        {
          "filename": "transfers.txt",
          "fieldName": "to_route_id",
          "index": 6
        }
      ]
    }
  ]
}
```


We need multiple `sampleNotices` to represent the several contexts of the notices described in the validation report. What do you think about renaming this attribute `sampleContexts` for more clarity? A `NoticeReport` would then have multiple `sampleContext`.

Reply:
I don't know, I think "contexts" is even less clear to someone reading the JSON and unfamiliar with the implementation.

I think it's ok to have the "sampleNotices" label in the JSON as long as the container class is named something different (e.g., `NoticeReport`) - I was just calling out the clash between having the container class named the same thing (but singular) as the internal data structure. "sampleNotices" seems like the most descriptive name for this info in the JSON (they are sample notices, sampled from the total notices collected).

Reply:
This is what I had in mind: `sampleNotices` renamed `contexts` . Having the following structure and using `@SerializedName` allows to have `sampleNotices` label in the JSON.

```java 
public class NoticeReport {

  @Expose() private final String code;
  @Expose() private final SeverityLevel severity;
  @Expose() private final int totalNotices;

  @SerializedName("sampleNotices")
  @Expose()
  private final List<LinkedTreeMap<String, Object>> contexts;
}
```

Reply:
I would always err on the side of naming things as descriptively as possible, and to me it seems like `sampleNotices` makes the most sense internally as well. `contexts` to me is unclear, and given that it's a map that stores generic Strings and Objects, I think further clarity is important. Just my two cents - you're closer to the code than I am, so it may make more sense to you in context (no pun intended :)) of the whole project.